src/data-structures/hash-table.js

```diff
@@ -41,25 +48,71 @@
 
     for (i = 0; i < str.length; i += 1) {
       character = str.charCodeAt(i);
+      /*jshint -W016 */
       hashCode = ((hashCode << 5) - hashCode) + character;
       hashCode = hashCode & hashCode;
+      /*jshint -W016 */
     }
 
     return hashCode;
   };
 
-  exports.HashTable.prototype.put = function (key, value) {
+  exports.HashTable.prototype.put = function (key, data) {
     var hashCode = this.hashCode(key);
-    this.elements[hashCode] = value;
+    var newNode = new Node(key, data);
+
+    if (this.elements[hashCode] === null) {
```

Comment:
Since `this.elements[hashCode]` might not be defined, I guess the check should be: `this.elements[hashCode] === undefined`